### ISSUE TITLE
Code Completion for Azure 

### ISSUE THREAD

**Issue by QuadStingray** [carlrobertoh/CodeGPT#435]
### Describe the need of your request

Code Completion sounds fantastic, and I would love to test it. When do you plan to activate this awesome feature for Azure OpenAI or Custom OpenAI Providers?

### Additional context

I like your Plugin and use it nearly every day

**Comment by PhilKes** [comment]
You should be able to use Azure for code-completions now (https://github.com/carlrobertoh/CodeGPT/pull/476)
In the Custom OpenAI Providers settings:
![image](https://github.com/carlrobertoh/CodeGPT/assets/39240633/54af4c14-c08f-4150-bb75-f7c3a52a13ea)

**Comment by FabGoe** [comment]
Thanks for the implemention. I tried the recent Azure GPT3 and GPT4 models, which only support the "chat/completations" API (not the normal "/completations" API). I cannot get it to run. The connection test fails (see screenshot).

![image](https://github.com/carlrobertoh/CodeGPT/assets/140496047/971bce44-fc7a-4ec8-8448-d493fb45753e)

**Comment by sisve** [comment]
That is correct, you would need to create a separate deployment using the gpt-3.5-turbo-instruct model to use the /completions api. Azure seems to ignore the model provided in the request body, and instead use the one associated with your deployment.

**Comment by FabGoe** [comment]
@sisve Thanks for your help and fast response! I got it working now with gpt-3.5-turbo-instruct. This wasn't documented well in the Azure OpenAI manuals.
